### PR TITLE
 #1178 attempt at adding in rpm epochs

### PR DESCRIPTION
**Pull request by Ophirr33**

### CODE REVIEW

File: src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala [hunk 1]
```diff
@@ -222,6 +223,9 @@ case class RpmSpec(meta: RpmMetadata,
     sb append ("Version: %s\n" format meta.version)
     sb append ("Release: %s\n" format meta.release)
     sb append ("Summary: %s\n" format meta.summary)
+
+    if (meta.epoch > 0) sb append ("Epoch: %d\n" format meta.epoch)
```
Comment (kardapoltsev) [inline src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala]: I think it's better to use `Option` in such cases

Comment (Ophirr33) [reply src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala]: I went back and forth on that. Doing this way also protests against negative epochs, but I suppose I could do an exists or filter for the negative check. Thoughts?

**Comment by kardapoltsev** [comment]
Yes, sounds good.

-- 
Regards, Alexey

File: src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala (hunk 1 above)
Comment (muuki88) [reply src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala]: That sounds good. Option + check.

You can also add a [validation check](https://github.com/sbt/sbt-native-packager/blob/master/src/main/scala/com/typesafe/sbt/packager/rpm/RpmPlugin.scala#L105) for this a log an error or warning.

Would `rpmLint` catch this?

File: integration-tests-ansible/test-project-play-rpm/packaging.sbt [hunk 2]
```diff
@@ -28,3 +28,5 @@ rpmRelease := "1"
 rpmVendor := "DemoVendor"
 
 rpmLicense := Some("Apache-2.0")
+
+rpmEpoch := 1
```
Comment (muuki88) [inline integration-tests-ansible/test-project-play-rpm/packaging.sbt]: Out of curiosity? This do you acutally run this test? I haven't used this for a very long time and I'm not sure if we should remove it.

Comment (Ophirr33) [reply integration-tests-ansible/test-project-play-rpm/packaging.sbt]: Whoops, I forgot that I committed this. I mucked around with the Vagrant file (e.g., changing chef to bento) and got something half running, but tbh you could probably remove it.

File: src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala (hunk 1 above)
Comment (Ophirr33) [reply src/main/scala/com/typesafe/sbt/packager/rpm/RpmMetadata.scala]: I don't think rpmLint has a check for this

**Comment by Ophirr33** [comment]
squashed and updated with review comments

**Comment by Ophirr33** [comment]
Can someone poke the travis build?

File: src/main/scala/com/typesafe/sbt/packager/validation/package.scala [hunk 3]
```diff
@@ -63,4 +63,16 @@ package object validation {
     }
   }
 
+  def epochIsNaturalNumber(epoch: Int): Validation.Validator = () => {
+    sys.error(s"Passed: $epoch")
```
Comment (kardapoltsev) [inline src/main/scala/com/typesafe/sbt/packager/validation/package.scala]: Did you forget to remove logging?

**Comment by kardapoltsev** [comment]
LGTM

**Comment by Ophirr33** [comment]
🥂

**Pull request merged**